modules/carto/src/layers/cluster-utils.ts

```diff
@@ -0,0 +1,159 @@
+import {cellToParent} from 'quadbin';
+import {_Tile2DHeader as Tile2DHeader} from '@deck.gl/geo-layers';
+import {Accessor} from '@deck.gl/core';
+import {BinaryFeatureCollection} from '@loaders.gl/schema';
+
+export type Aggregation = 'any' | 'average' | 'count' | 'min' | 'max' | 'sum';
+export type AggregationProperties<FeaturePropertiesT> = {
+  aggregation: Aggregation;
+  name: keyof FeaturePropertiesT;
+}[];
+export type ClusteredFeaturePropertiesT<FeaturePropertiesT> = FeaturePropertiesT & {
+  id: bigint;
+  count: number;
+  position: [number, number];
+  stats: Record<keyof FeaturePropertiesT, {min: number; max: number}>;
+};
+export type ParsedQuadbinCell<FeaturePropertiesT> = {id: bigint; properties: FeaturePropertiesT};
+export type ParsedQuadbinTile<FeaturePropertiesT> = ParsedQuadbinCell<FeaturePropertiesT>[];
+
+export function aggregateTile<FeaturePropertiesT>(
+  tile: Tile2DHeader<ParsedQuadbinTile<FeaturePropertiesT>>,
+  aggregationLevels: number,
+  properties: AggregationProperties<FeaturePropertiesT> = [],
+  getPosition: Accessor<ParsedQuadbinCell<FeaturePropertiesT>, [number, number]>,
+  getWeight: Accessor<ParsedQuadbinCell<FeaturePropertiesT>, number>
+): void {
+  if (!tile.content) return;
+
+  // Aggregate on demand and cache result
+  if (!tile.userData) tile.userData = {};
+  if (tile.userData[aggregationLevels]) return;
+
+  const out: Record<number, any> = {};
+  for (const cell of tile.content) {
+    let id = cell.id;
+    const position = typeof getPosition === 'function' ? getPosition(cell, {} as any) : getPosition;
+
+    // Aggregate by parent id
+    for (let i = 0; i < aggregationLevels - 1; i++) {
+      id = cellToParent(id);
+    }
+
+    // Unfortunately TS doesn't support Record<bigint, any>
+    // https://github.com/microsoft/TypeScript/issues/46395
+    const parentId = Number(id);
```

Comment:
Out of curiosity, are we using bigint because we might need the larger values? Or mostly as a way to enforce integer types?

Reply:
[The quadbin index is a bigint](https://github.com/visgl/deck.gl/blob/master/modules/carto/src/layers/schema/spatialjson-utils.ts#L6)